### PR TITLE
9195 New coreadm config causes failure in zpool_003_pos

### PR DESCRIPTION
**Pull request by brad-lewis**
Reviewed by: Serapheim Dimitropoulos <serapheim@delphix.com>
Reviewed by: Dan Kimmel dan.kimmel@delphix.com

If a core file is not created in the cwd the test fails. The solution is
to invoke `coreadm -i core` just before running this test.

Also included are Makefile changes in zfstest cannot cope with empty
directories.

Upstream bugs: DLPX-51761, DLPX-51770

### CODE REVIEW

File: usr/src/test/zfs-tests/cmd/scripts/zfstest.ksh [hunk 1]
```diff
@@ -176,6 +176,8 @@ export KEEP="^$(echo $KEEP | sed 's/ /$|^/g')\$"
 num_disks=$(echo $DISKS | awk '{print NF}')
 [[ $num_disks -lt 3 ]] && fail "Not enough disks to run ZFS Test Suite"
 
+sudo -k coreadm -e process
```
Comment (yuripv) [inline usr/src/test/zfs-tests/cmd/scripts/zfstest.ksh]: This effectively changes coreadm settings and doesn't restore them, which makes this change a bit pointless -- we could just mention the needed settings in README instead.

File: usr/src/test/zfs-tests/Makefile [hunk 2]
```diff
@@ -15,6 +15,6 @@
 
 .PARALLEL: $(SUBDIRS)
 
-SUBDIRS:sh = find ./* -maxdepth 0 -type d
+SUBDIRS:sh = ls */Makefile 2>/dev/null | sed 's/\/Makefile//g'
```
Comment (yuripv) [inline usr/src/test/zfs-tests/Makefile]: This doesn't seem to be related?

File: usr/src/test/zfs-tests/Makefile.com [hunk 3]
```diff
@@ -60,4 +60,4 @@ $(TARGETDIR)/%: %
 	$(INS.file)
 
 .PARALLEL: $(SUBDIRS)
-SUBDIRS:sh = find ./* -maxdepth 0 -type d
+SUBDIRS:sh = ls */Makefile 2>/dev/null | sed 's/\/Makefile//g'
```
Comment (yuripv) [inline usr/src/test/zfs-tests/Makefile.com]: This doesn't seem to be related?

**Comment by ahrens** [comment]
@jwk404 could you take a look at @yuripv's comments?

File: usr/src/test/zfs-tests/cmd/scripts/zfstest.ksh (hunk 1 above)
Comment (ikozhukhov) [reply usr/src/test/zfs-tests/cmd/scripts/zfstest.ksh]: i think will be better to save current state -> update it for tests -> restore it
no need just point it in README

Comment (yuripv) [reply usr/src/test/zfs-tests/cmd/scripts/zfstest.ksh]: Or that, I just don't like system-wide settings being modified without me knowing about it.

BTW, the change itself is missing the ```-i core```, which was in the description.

Comment (jwk404) [reply usr/src/test/zfs-tests/cmd/scripts/zfstest.ksh]: The `-i core` from the description was changed during review - the description is wrong.
I think it's better to have this command in the wrapper script than have a comment in the README that folks may or may not read. I also feel that it's appropriate to change system wide settings on a test system.
That said, if this is a blocker for you, let us know; it's easy enough to restore the setting. Alternatively, we could simply abandon this PR - I don't expect that coreadm settings need to be changed on other platforms.

**Comment by prakashsurya** [comment]
Based on the discussion between @yuripv @ikozhukhov and @jwk404, I'm going to close this out. It sounds like it's not worth pushing this change outside of Delphix.

**Comment by yuripv** [comment]
I'm sorry this didn't work out, but it really looks like a distribution-specific configuration issue.

**Comment by prakashsurya** [comment]
Thanks @yuripv, no worries! Sorry for the noise.
